storage/engine/engine.go

```diff
@@ -87,17 +87,14 @@ type Iterator interface {
 	ComputeStats(start, end MVCCKey, nowNanos int64) (MVCCStats, error)
 }
 
-// Engine is the interface that wraps the core operations of a
-// key/value store.
-type Engine interface {
-	// Open initializes the engine.
-	Open() error
-	// Close closes the engine, freeing up any outstanding resources.
+// Reader is the read interface to an engine's data.
+type Reader interface {
+	// Close closes the reader, freeing up any outstanding resources.
 	Close()
-	// Attrs returns the engine/store attributes.
-	Attrs() roachpb.Attributes
-	// Put sets the given key to the value provided.
-	Put(key MVCCKey, value []byte) error
+	// closed returns true if the reader has been closed or is not usable.
```

Comment:
These all seem to have pretty clear parallels to the `io` package, so if this method ever needs to be exported, it should probably be pulled into a `Closer` interface.